src/main/java/frc/robot/Robot.java

```diff
@@ -238,89 +166,83 @@ public void robotPeriodic() {
 
 
   public void teleopPeriodic() {
-
+    // read shuffleboard inputs
     velSetpointsShuffleboard();
     headingSetpointsShuffleboard();
 
-    double a = m_driveEncoderA.getVelocity();
-    double b = m_driveEncoderB.getVelocity();
+    // get the motor speeds
+    double aMotorRPM = m_driveEncoderA.getVelocity();
+    double bMotorRPM = m_driveEncoderB.getVelocity();
 
-    double actualDriveRPM = getWheelVelocity(a, b);
-    double actualYawRPM   = getModuleYaw(a, b);
+    // calculate the wheel speeds from those motor speeds
+    double currentWheelRpm      = motorRPMsToWheelRPM(aMotorRPM, bMotorRPM);
+    double currentModuleYawRPM  = motorRPMsToModuleYawRPM(aMotorRPM, bMotorRPM);
 
     // report to dashboard
-    sb_rot.setDouble(actualYawRPM);
-    sb_trans.setDouble(actualDriveRPM);
+    sb_rot.setDouble(currentModuleYawRPM);
+    sb_trans.setDouble(currentWheelRpm);
 
     // step 1 - tune the motor internal PIDs
-    if (sb_use_volts.getBoolean(false)) {
-      double aVolts = sb_a_volts.getDouble(0);
-      double bVolts = sb_b_volts.getDouble(0);
+    // allow overriding desired motor speeds during running test
+    if (sb_use_motorRPM.getBoolean(false)) {
+      double aDesiredMotorRPM = sb_a_motorRPM.getDouble(0);
+      double bDesiredMotorRPM = sb_b_motorRPM.getDouble(0);
      
-      m_aPID.setReference(aVolts, ControlType.kVelocity);
-      m_bPID.setReference(bVolts, ControlType.kVelocity);
+      m_aPID.setReference(aDesiredMotorRPM, ControlType.kVelocity);
+      m_bPID.setReference(bDesiredMotorRPM, ControlType.kVelocity);
 
-       // report to dashboard
-      sb_voltsA.setDouble(-1);
-      sb_voltsB.setDouble(-1);
       return;
     }
     
-    double desiredYawAngle = sb_heading.getDouble(0);  //degrees, 0=from home to down-field; degrees increase clockwise
-    double desiredDriveRPM = sb_velocity.getDouble(0); //input is [-1,1]
+    // Get desired yaw and wheel speed
+    double desiredYawDeg = sb_heading.getDouble(0);
+    double desiredWheelRPM = sb_velocity.getDouble(0);
 
-    // double calcTurnVelocity  = 
-    // double calcDriveVelocity = desiredDriveVelocity; //pid_drive.calculate(actualDriveVelocity, desiredDriveVelocity) + driveFF;
+    // double pidCalculatedWheelRPM = pid_wheel.calculate(currentWheelRpm, desiredWheelRPM) + driveFF;
+    double pidCalculatedWheelRPM  = desiredWheelRPM; // TODO: OVERRIDES LINE ABOVE!
 
-    double yaw_pid_calc = pid_yaw.calculate(getHeadingDeg(), desiredYawAngle);
+    // TODO: This looks wrong. We obtain a heading in degrees from
+    // shuffleboard (desiredYawDeg) and the current module heading in
+    // degrees from the the encoder, and then expect to get back an
+    // RPM?????
```

Comment:
PID Magic ... we are using the PID to approximate the function. More testing needed